src/main/java/org/springframework/data/redis/connection/jedis/StreamConverters.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2018-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.redis.connection.jedis;
+
+import org.springframework.data.redis.connection.stream.ByteRecord;
+import org.springframework.data.redis.connection.stream.Consumer;
+import org.springframework.data.redis.connection.stream.PendingMessage;
+import org.springframework.data.redis.connection.stream.PendingMessages;
+import org.springframework.data.redis.connection.stream.RecordId;
+import org.springframework.data.redis.connection.stream.StreamRecords;
+import redis.clients.jedis.BuilderFactory;
+import redis.clients.jedis.util.SafeEncoder;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+/**
+ * Converters for Redis Stream-specific types.
+ * <p/>
+ * Converters typically convert between value objects/argument objects retaining the actual types of values (i.e. no
+ * serialization/deserialization happens here).
+ *
+ * @author dengliming
+ * @since 2.3
+ */
+@SuppressWarnings({ "rawtypes" })
+class StreamConverters {
+
+	private static final BiFunction<Object, String, org.springframework.data.redis.connection.stream.PendingMessages> PENDING_MESSAGES_CONVERTER = (
```

Comment:
We refactored a few converters recently to do the conversion directly in the actual conversion method (e.g. `toPendingMessages`) as the indirection over `BiFunction` or `Converter` always requires navigating to the actual implementation. Having the conversion code in the actual method makes the code easier to read.

Reply:
Thanks. see the lastest commit.